src/main/java/com/minecolonies/coremod/colony/Colony.java

```diff
@@ -948,7 +949,14 @@ public void spawnCitizen(final CitizenData data)
             CitizenData citizenData = data;
             if (citizenData == null)
             {
-                topCitizenId++;
+                for (int i = 1; i <= MAX_CITIZENS_COUNTER; i++)
+                {
+                    if (getCitizen(i) == null)
```

Comment:
We should add a comment explaining that we ensure this way, that all Citizen ids get reused. ANd maybe explain why this is needed